source/common/http/conn_manager_impl.h

```diff
@@ -334,44 +338,42 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
 
     // All state for the stream. Put here for readability.
     struct State {
-      State()
-          : codec_saw_local_complete_(false), codec_encode_complete_(false),
-            on_reset_stream_called_(false), is_zombie_stream_(false), successful_upgrade_(false),
-            is_internally_destroyed_(false), is_internally_created_(false), is_tunneling_(false),
-            decorated_propagate_(true), deferred_to_next_io_iteration_(false),
-            deferred_end_stream_(false) {}
-
       // It's possibly for the codec to see the completed response but not fully
       // encode it.
-      bool codec_saw_local_complete_ : 1; // This indicates that local is complete as the completed
-                                          // response has made its way to the codec.
-      bool codec_encode_complete_ : 1;    // This indicates that the codec has
-                                          // completed encoding the response.
-      bool on_reset_stream_called_ : 1;   // Whether the stream has been reset.
-      bool is_zombie_stream_ : 1;         // Whether stream is waiting for signal
-                                          // the underlying codec to be destroyed.
-      bool successful_upgrade_ : 1;
```

Comment:
These changes mainly used to fix clang.tidy.

Reply:
Hm. I think that removing the `: 1` means that each of these fields will get 8x larger. What's the clang tidy error?

Reply:
```
Use default member initializer for 'codec_saw_local_complete_' (fix available)clang-tidy(modernize-use-default-member-init)
```

Reply:
Hm. I think we'd prefer to avoid making this struct larger if we don't need to. Can you do a `sizeof(State)` before and after this change to see what it's doing? I wonder if we could use both `: 1` and ` = false` to get the best of both worlds?

Reply:
Yeah, you are right, it make the state memory footprint from 2 bytes to 11 bytes.

Although I am not sure if that make sense or not for software like envoy. we never pursue extreme low memory overhead.

But I still revert all these changes. I think @phlax is working on clang-tidy related things. we can do all these at there or change our clang-tidy rule.